packages/perseus/src/widgets/cs-program/cs-program.tsx

```diff
@@ -130,14 +108,11 @@ class CSProgram extends React.Component<Props> {
         return Changeable.change.apply(this, args);
     };
 
-    simpleValidate(rubric): PerseusScore {
-        return CSProgram.validate(
-            {
-                status: this.props.status,
-                message: this.props.message,
-            },
-            rubric,
-        );
+    simpleValidate(): PerseusScore {
+        return csProgramValidator({
+            status: this.props.status,
+            message: this.props.message,
```

Comment:
The data we pass to the `csProgramValidator` should be extracted into a [`getUserInput` function](https://github.com/Khan/perseus/blob/f5af2437133d68dcbaa42830850341c46d7affee/packages/perseus/src/renderer.tsx#L148). This will help us as we split the scoring logic further out of the renderer system. 

You can see this pattern in play in many of our widgets (eg [`grapher`](https://github.com/Khan/perseus/blob/f326139ee6bf9680075eac6353b1e84c44fa3f77/packages/perseus/src/widgets/grapher/grapher.tsx#L544) or [`number-line`](https://github.com/Khan/perseus/blob/ae51ccdb820894f6fc5c1b23556823efdd4edba6/packages/perseus/src/widgets/number-line/number-line.tsx#L638))

Could you do this in a new PR?

Reply:
Here's the ticket for that: https://khanacademy.atlassian.net/browse/LEMS-2459